ArcGISRuntimeSDKQt_CppSamples/Features/ListRelatedFeatures/ListRelatedFeatures.cpp

```diff
@@ -80,58 +91,56 @@ void ListRelatedFeatures::connectSignals()
   {
     if (!loadError.isEmpty())
       return;
-    for (int i = 0; i < m_map->operationalLayers()->size(); i++)
+
+    for (int i = 0; i < m_map->operationalLayers()->size(); ++i)
     {
       // get the Alaska National Parks layer
-      if (m_map->operationalLayers()->at(i)->name().contains(QString("- Alaska National Parks")))
+      if (m_map->operationalLayers()->at(i)->name().contains(QStringLiteral("- Alaska National Parks")))
       {
         m_alaskaNationalParks = static_cast<FeatureLayer*>(m_map->operationalLayers()->at(i));
+        m_alaskaFeatureTable = static_cast<ArcGISFeatureTable*>(m_alaskaNationalParks->featureTable());
+
+        // connect to queryRelatedFeaturesCompleted signal
+        connect(m_alaskaFeatureTable, &ArcGISFeatureTable::queryRelatedFeaturesCompleted,
+                this, [this](QUuid, QList<RelatedFeatureQueryResult*> rawRelatedResults)
+                {
+                  FeatureQueryListResultLock lock(rawRelatedResults);
+                  for (const RelatedFeatureQueryResult* relatedResult : lock.results)
+                  {
+                    while (relatedResult->iterator().hasNext())
+                    {
+                      // get the related feature
+                      ArcGISFeature* feature = static_cast<ArcGISFeature*>(relatedResult->iterator().next());
+                      ArcGISFeatureTable* relatedTable = static_cast<ArcGISFeatureTable*>(feature->featureTable());
+                      QString displayFieldName = relatedTable->layerInfo().displayFieldName();
+                      QString serviceLayerName = relatedTable->layerInfo().serviceLayerName();
+                      QString displayFieldValue = feature->attributes()->attributeValue(displayFieldName).toString();
+
+                      // add the related feature to the list model
+                      RelatedFeature relatedFeature = RelatedFeature(displayFieldName,
+                                                                     displayFieldValue,
+                                                                     serviceLayerName);
+                      m_relatedFeaturesModel->addRelatedFeature(relatedFeature);
+                      emit relatedFeaturesModelChanged();
+                    }
+                  }
+                  emit showAttributeTable();
+                });
 
         // connect to selectFeaturesCompleted signal
         connect(m_alaskaNationalParks, &FeatureLayer::selectFeaturesCompleted, this, [this](QUuid, FeatureQueryResult* rawResult)
         {
-          QPointer<FeatureQueryResult> result(rawResult);
+          QScopedPointer<FeatureQueryResult> result(rawResult);
           // iterate over features returned
           while (result->iterator().hasNext())
```

Comment:
It's kind of weird this is a while loop. I presume there is usually only 1... but the way its written we would query for related features and then zoom to each feature in turn.